src/realm/sync/client.cpp

```diff
@@ -405,7 +405,9 @@ ClientImpl::~ClientImpl()
 void ClientImpl::cancel_reconnect_delay()
 {
     // Thread safety required
-    auto handler = [this] {
+    post([this](Status status) {
+        if (status == ErrorCodes::OperationAborted)
```

Comment:
for these places where we check if the status is operation aborted, should we also REALM_ASSERT that the status.is_ok() before continuing to the function body? We can do this in a separate PR since right now we know that OperationAborted is the only error code likely to be passed in here, but error checking in these callbacks is probably something to tighten up overall.

Reply:
For now, I'm throwing an exception if status is an unexpected value, since I'm not sure if `this` or the logger is still valid when this is called.